tck/src/main/java/org/eclipse/microprofile/lra/tck/participant/api/ContextTckResource.java

```diff
@@ -0,0 +1,377 @@
+                          @PathParam("metric") String metric) {
+        if (metrics.containsKey(lraId)) {
+            return Response.ok(metrics.get(lraId).get(metric)).build();
+        }else {
+            return Response.ok(-1).build();
+        }
+    }
+
+    @Leave
+    @PUT
+    @Path(LEAVE_PATH)
+    public Response leave(@HeaderParam(LRA_HTTP_CONTEXT_HEADER) String lraId) {
+        return Response.ok().entity(lraId).build();
+    }
+
+    @PUT
+    @Path("/compensate")
+    @Compensate
+    public Response compensateWork(@HeaderParam(LRA_HTTP_CONTEXT_HEADER) String lraId)
+            throws NotFoundException {
+        incrementMetric(Compensate.class.getName(), lraId);
+
+        return getEndPhaseResponse(false);
+    }
+
+    @PUT
+    @Path("/complete")
+    @Complete
+    public Response completeWork(@HeaderParam(LRA_HTTP_CONTEXT_HEADER) String lraId)
+            throws NotFoundException {
+        incrementMetric(Complete.class.getName(), lraId);
+
+        return getEndPhaseResponse(true);
+    }
+
+    @Status
+    @GET
+    @Path(STATUS_PATH)
+    public Response status(@HeaderParam(LRA_HTTP_CONTEXT_HEADER) String lraId) {
+        incrementMetric(Status.class.getName(), lraId);
+
+        return Response.status(endPhaseStatus).entity(status.name()).build();
+    }
+
+    @Forget
+    @DELETE
+    @Path("/forget")
+    public Response forget(@HeaderParam(LRA_HTTP_CONTEXT_HEADER) String lraId) {
+        incrementMetric(Forget.class.getName(), lraId);
+
+        return Response.status(endPhaseStatus).entity(status.name()).build();
+    }
+
+    // clear any injected participant behaviour
+    @PUT
+    @Path(STATUS_PATH)
+    public Response clearStatus(@HeaderParam(LRA_TCK_HTTP_CONTEXT_HEADER) String lraId,
+                           @HeaderParam(LRA_TCK_FAULT_TYPE_HEADER) String tckFaultType) {
+        switch (status) {
+            case Compensating:
+                status = ParticipantStatus.Compensated;
+                break;
+            case Completing:
+                status = ParticipantStatus.Completed;
+                break;
+            default:
+                break;
+        }
+
+        endPhase = EndPhase.SUCCESS;
+        endPhaseStatus = Response.Status.OK;
+
+        return Response.ok().entity(status.name()).build();
+    }
+
+    // modify participant responses based on injected behavour specified by the test
+    private Response getEndPhaseResponse(boolean completing) {
+        switch (endPhase) {
+            case ACCEPTED:
+                status = completing ? ParticipantStatus.Completing : ParticipantStatus.Compensating;
+                return Response.status(Response.Status.ACCEPTED).entity(status.name()).build();
+            case FAILED:
+                status = completing ? ParticipantStatus.FailedToComplete : ParticipantStatus.FailedToCompensate;
+                return Response.status(endPhaseStatus).entity(status.name()).build();
+            case SUCCESS: /* FALLTHRU */
```

Comment:
I always find a default clause something dangerous as we might miss this with a future update.

Reply:
The alternative is to duplicate the code in the SUCCESS branch after the switch which is probably more error prone than omitting the default branch.